src/store_rebuild.cc

```diff
@@ -325,38 +276,26 @@ storeRebuildParseEntry(MemBuf &buf, StoreEntry &tmpe, cache_key *key,
                        StoreRebuildData &stats,
                        uint64_t expectedSize)
 {
-    int swap_hdr_len = 0;
-    StoreMetaUnpacker aBuilder(buf.content(), buf.contentSize(), &swap_hdr_len);
-    if (aBuilder.isBufferZero()) {
-        debugs(47,5, "skipping empty record.");
-        return false;
-    }
+    uint64_t swap_hdr_len = 0;
+
+    tmpe.key = nullptr;
 
-    StoreMeta *tlv_list = nullptr;
     try {
-        tlv_list = aBuilder.createStoreMeta();
-    } catch (const std::exception &e) {
-        debugs(47, DBG_IMPORTANT, "WARNING: Ignoring store entry because " << e.what());
+        swap_hdr_len = Store::UnpackIndexSwapMeta(buf, tmpe, key);
+    } catch (...) {
+        debugs(47, Important(65), "WARNING: Indexer ignores a cache_dir entry: " << CurrentException);
```

Comment:
I added `cache_log_message` control to this WARNING because I see that PR-unrelated problems in rock and ufs indexing code will trigger this warning in some cases. In certain environments, there can be many such warnings. It may take us a little bit of time to fix those (out of scope) problems in old cache_dir indexing code, so I thought it would be a good idea to get early adopters (and testers) more control over this warning.

Even after those bugs are fixed, this control may be useful in environments that simply have to deal with dirty cache_dir inputs (e.g., due to frequent abrupt Squid restarts).

Said that, I will undo without arguing if there are any objections to placing this WARNING under `cache_log_message` control.

Reply:
LGTM.